### DISCUSSION TITLE
RDZsonde isn't working with non-broadcasting SSID

### DISCUSSION THREAD

**Discussion by pamidev** [dl9rdz/rdz_ttgo_sonde#278] [TTGO firmware]
Hello.

I noticed that rdzTTGOsonde (last devel issues) does not connect to a hidden WiFi network (non broadcasting SSID). I checked TTGO boards: T-beam v.1.1 and Lora32 v2.1_1.6. If WiFi is visible, there isn't problem. Can anybody confirm that? Is this a known problem and fixable?

73, pamidev

**Comment by py2oc** [comment]
Having the same issue here. The board connects only to "visible" ssid networks.
If anyone know how to fix that issue, please let us know.

73 de PY2OC, Luiz

**Comment by hr-ru** [comment]
That is the way how it is implemented. It scans for networks (which only finds visible networks), and then if the scan returns one of the networks in the config, connects to that network.

You can "fix" that in the code by removing the scan part. 
For Wifi mode 3, the simplest way is changing RX_FSK.ini around line 2891.
Replace all the code between
`WiFi.mode(WIFI_STA);`
and
`int lastl = (disph / dispys - 2) * dispys;`
with a single line
`int index = 1;`

For Wifi mode 1, a different part needs to be change. Probably best in loopWifiBackground, replace
```
    wifi_state = WIFI_SCAN;
    Serial.println("WiFi start scan");
    WiFi.scanNetworks(true); // scan in async mode
```
with
```
   wifi_state = WIFI_CONNECT;
   WiFi.begin(fetchWifiSSID(1), fetchWifiPw(1));
```

**Comment by dl9rdz** [comment]
New devel version adds a WiFi mode 4, which does exactly this.
